### DISCUSSION TITLE
What causes a script name to have a red line through it?

### DISCUSSION THREAD

**Discussion by TFWol** [violentmonkey/violentmonkey#1744] [Q&A] [answered]
This occurs in the script drop-down menu.
I might be overlooking it, but I can't find a reference to what this means, let alone fix it.

Could someone point me in the right direction?

**Comment by tophf** [comment, marked as answer]
It usually means a syntax error in the script. Unfortunately such errors cannot be detected so we can't show the actual error, but now it occurred to me we should add some kind of explanation at least in a hover tooltip.

Another reason is a mismatch of `@inject-into` mode. If the script specifies `page` it may fail in Firefox on sites that define a strict CSP or on sites that try to prevent userscript injection (this is extremely rare, I haven't see one myself). The mode can be also changed globally by choosing `page` inject mode in advanced settings.

**Reply by TFWol** [reply]
Awesome, thank you very much for the reply!

**Reply by Omaeda** [reply]
> It usually means a syntax error in the script. Unfortunately such errors cannot be detected so we can't show the actual error, but now it occurred to me we should add some kind of explanation at least in a hover tooltip.

I have this same problem. Is there any way to find out what specifically is causing the error? is the error being displayed somewhere? is there a debug mode for this case?

**Reply by tophf** [reply]
If it's a syntax error, it'll be displayed in the devtools console.
